packages/costa/src/runtime.ts

```diff
@@ -42,11 +42,22 @@ function selectNpmPackage(metadata: NpmPackageMetadata, source: PluginSource): N
   return metadata.versions[metadata['dist-tags'].latest];
 }
 
-function spawnAsync(command: string, args?: string[], opts: SpawnOptions = {}): Promise<string> {
+interface LogWriter {
+  stdout: Writable;
+  stderr: Writable;
+}
+
+interface CostaSpawnOptions extends SpawnOptions {
+  logWriter: LogWriter;
+}
```

Comment:
This could be simplified in:

```ts
type CostaSpawnOptions = { stdout: Writable, stderr: Writable } & SpawnOptions;
```

Reply:
Ping @FeelyChau

Reply:
I prefer to pack them to a object, they are used for log.